nexus/src/external_api/http_entrypoints.rs

```diff
@@ -2449,24 +2609,96 @@ async fn disk_delete(
     let apictx = rqctx.context();
     let nexus = &apictx.nexus;
     let path = path_params.into_inner();
-    let organization_name = &path.organization_name;
-    let project_name = &path.project_name;
-    let disk_name = &path.disk_name;
+    let disk_selector = params::DiskSelector::new(
+        Some(path.organization_name.into()),
+        Some(path.project_name.into()),
+        path.disk_name.into(),
+    );
     let handler = async {
         let opctx = OpContext::for_external_api(&rqctx).await?;
-        nexus
-            .project_delete_disk(
-                &opctx,
-                &organization_name,
-                &project_name,
-                &disk_name,
-            )
-            .await?;
+        let disk_lookup = nexus.disk_lookup(&opctx, &disk_selector)?;
+        nexus.project_delete_disk(&disk_lookup).await?;
         Ok(HttpResponseDeleted())
     };
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+/// Attach a disk to an instance
+#[endpoint {
+    method = POST,
+    path = "/v1/disks/{disk}/attach",
+    tags = ["disks"],
+}]
```

Comment:
> **Warning**
> This section warrants extra attention. 

With attach and detach actions I took the same sort of approach as I did with listing in that I collapsed it down to the disks endpoint. 

Before:
- `POST /.../instances/{instance}/disks/attach { disk: Name }`
- `POST /.../instances/{instance}/disks/detach { disk: Name}`

After
- `/v1/disks/attach { instance: NameOrId }`
- `/v1/disks/detach { instance: NameOrId }`

Given the overall flatter structure of the API I think this mapping makes slightly more sense. Input welcome.

Reply:
Unfortunately I don't think this approach will be as clear as the previous one. You generally attach a disk to an instance(vm), and not the other way around. Flipping it around can potentially confuse people and the endpoint itself could be misleading. 

What do you think about something like: `/v1/instance/{instance}/attach { disk: NameOrId }` instead of `/v1/disks/{disk}/attach { instance: NameOrId }`

We can even go a bit further by having a very explicit endpoint like: `/v1/instance/{instance}/attach-disk { disk: NameOrId }`

Reply:
I see what you're saying. The challenge with attach and detach is it's a two way action. It acts on both the disk and the instance. The benefit of the implementation I have is that there's only a single place to look for disks within the API. If clarification is the main sticking point it could just as well be updated to be `/v1/disks/{disk}/attach-to-instance` and `/v1/disks/{disk}/detach-from-instance`. If we go back to hanging it off of instances then I believe we should keep with the previous model of `/instance/{instance}/disks/*`.

Reply:
I'm going to roll this back into the original style. I found an issue in my implementation that would require updates at the Nexus app layer to address which I'd rather not do.

Reply:
Okay, this is reverted to

- `/v1/instances/{instance}/disks/attach`
- `/v1/instances/{instance}/disks/detach`

Reply:
Agreed with where you landed on this: I'm "talking to" the instance and the object of the attach/detach "verb" is the disk.